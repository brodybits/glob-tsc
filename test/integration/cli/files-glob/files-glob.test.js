const execa = require('execa');

const path = require('path');

// THANKS:
// https://stackoverflow.com/questions/49603939/async-callback-was-not-invoked-within-the-5000ms-timeout-specified-by-jest-setti/49864436#49864436
jest.setTimeout(10*1000);

// THANKS for initial guidance:
// * https://medium.com/@ole.ersoy/unit-testing-commander-scripts-with-jest-bc32465709d6
// * https://github.com/superflycss/cli/blob/master/index.spec.js
// * https://github.com/brodybits/create-react-native-module/blob/0.11.0/tests/integration/cli/help/cli-help.test.js
test('bin/glob-tsc.js --files-glob', async () => {
    const { stdout } = await execa.command(`node ${path.resolve('bin/glob-tsc.js')} --allowJs --checkJs --noEmit --files-glob test/integration/cli/files-glob/demo/demo.js`);

    expect(stdout).toMatchSnapshot();
});
