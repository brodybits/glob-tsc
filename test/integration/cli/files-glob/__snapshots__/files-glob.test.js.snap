// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bin/glob-tsc.js --files-glob 1`] = `
"checking for valid \`tsc\` CLI program
Version 3.6.4
valid \`tsc\` CLI program found"
`;
